src/configure/templateInputHelper/InputControlProvider.ts

```diff
@@ -51,110 +69,91 @@ export class InputControlProvider {
         for (let input of this._pipelineTemplate.inputs) {
             let inputControl: InputControl = null;
             let inputControlValue = this._getInputControlValue(input);
-
-            switch (input.inputMode) {
-                case InputMode.None:
-                case InputMode.AzureSubscription:
-                    if (input.type !== InputDataType.Authorization) {
-                        inputControl = new InputControl(input, inputControlValue, ControlType.None, this.azureSession);
-                    }
-                    break;
-                case InputMode.TextBox:
-                case InputMode.PasswordBox:
-                    inputControl = new InputControl(input, inputControlValue, ControlType.InputBox, this.azureSession);
-                    break;
-                case InputMode.Combo:
-                case InputMode.CheckBox:
-                case InputMode.RadioButtons:
-                    inputControl = new InputControl(input, inputControlValue, ControlType.QuickPick, this.azureSession);
-                    break;
-
-            }
-            if (inputControl) {
-                this._inputControlsMap.set(input.id, inputControl);
+            if (input.type !== InputDataType.Authorization) {
+                let controlType: ControlType = InputControlUtility.getInputControlType(input.inputMode);
+                inputControl = new InputControl(input, inputControlValue, controlType, this.azureSession);
+                if (inputControl) {
+                    this._inputControlsMap.set(input.id, inputControl);
+                    this._setInputControlDataSourceInputs(inputControl);
```

Comment:
this should aslo be called after overrideParameters in getAllPipelineTemplateInputs()

Reply:
updated